File: Módulo Python/Códigos/4 - Estrutura de Repetição/Exercicio.37.py
#37. Agora repita o exercício anterior para um número indefinido de pessoas.

total_salario_2_filhos = 0.0
contador_2_filhos = 0
total_salario_sem_filhos = 0.0
contador_sem_filhos = 0
total_salario_1_filho = 0.0
contador_1_filho = 0
total_salario = 0.0
contador_geral = 0

numeroVezes = int(input("Digite quantas pessoas deseja adicionar: "))

for _ in range(numeroVezes):
    nome = input("Digite o nome da pessoa: ")
    try:
        salario = float(input("\nDigite o salário da pessoa: "))
        numero_filhos = int(input("Digite o número de filhos da pessoa: "))
    except ValueError:
        print("Entrada inválida. Ignorando esta pessoa.")
        continue  
        
    total_salario += salario
    contador_geral += 1
    
    if numero_filhos == 2:
        total_salario_2_filhos += salario
        contador_2_filhos += 1
    elif numero_filhos == 0:
        total_salario_sem_filhos += salario
        contador_sem_filhos += 1
    elif numero_filhos == 1:
        total_salario_1_filho += salario
        contador_1_filho += 1

media_salario_2_filhos = total_salario_2_filhos / contador_2_filhos if contador_2_filhos > 0 else 0
media_salario_sem_filhos = total_salario_sem_filhos / contador_sem_filhos if contador_sem_filhos > 0 else 0
media_salario_1_filho = total_salario_1_filho / contador_1_filho if contador_1_filho > 0 else 0
media_salario_geral = total_salario / contador_geral if contador_geral > 0 else 0

print(f"Salário médio das pessoas com 2 filhos: {media_salario_2_filhos:.2f}")
print(f"Salário médio das pessoas sem filhos: {media_salario_sem_filhos:.2f}")
print(f"Média salarial maior entre pessoas com 1 e 2 filhos: {max(media_salario_1_filho, media_salario_2_filhos):.2f}")
print(f"Salário médio geral: {media_salario_geral:.2f}")

File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio37.java
// Agora repita o exercício anterior para um número indefinido de pessoas

import java.util.Scanner;

public class Exercicio37 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int pessoasComDoisFilhos = 0;
        int pessoasSemFilhos = 0;
        int pessoasComUmFilho = 0;

        double salarioTotal = 0;
        double salarioComDoisFilhos = 0;
        double salarioSemFilhos = 0;
        double salarioComUmFilho = 0;

        while (true) {
            System.out.println("Digite os dados da pessoa ou digite 'sair' para encerrar:");
            System.out.print("Nome: ");
            String nome = scanner.next();

            if (nome.equalsIgnoreCase("sair")) {
                break;
            }

            System.out.print("Salário: ");
            double salario = scanner.nextDouble();
            salarioTotal += salario;

            System.out.print("Número de filhos: ");
            int numeroFilhos = scanner.nextInt();

            if (numeroFilhos == 2) {
                pessoasComDoisFilhos++;
                salarioComDoisFilhos += salario;
            } else if (numeroFilhos == 0) {
                pessoasSemFilhos++;
                salarioSemFilhos += salario;
            } else if (numeroFilhos == 1) {
                pessoasComUmFilho++;
                salarioComUmFilho += salario;
            }
        }

        scanner.close();

        // Cálculo das médias
        double mediaSalarioComDoisFilhos = pessoasComDoisFilhos > 0 ? salarioComDoisFilhos / pessoasComDoisFilhos : 0;
        double mediaSalarioSemFilhos = pessoasSemFilhos > 0 ? salarioSemFilhos / pessoasSemFilhos : 0;
        double mediaSalarioComUmFilho = pessoasComUmFilho > 0 ? salarioComUmFilho / pessoasComUmFilho : 0;
        double mediaSalarioGeral = salarioTotal / (pessoasComDoisFilhos + pessoasSemFilhos + pessoasComUmFilho);

        System.out.println("\nResultados:");
        System.out.println("Média salarial das pessoas com 2 filhos: " + mediaSalarioComDoisFilhos);
        System.out.println("Média salarial das pessoas sem filhos: " + mediaSalarioSemFilhos);
        System.out.println("Média salarial maior entre os que têm um e dois filhos: " + Math.max(mediaSalarioComUmFilho, mediaSalarioComDoisFilhos));
        System.out.println("Média salarial geral: " + mediaSalarioGeral);
    }
}